### PR TITLE
Docstrings API examples

### LINKED ISSUE #596: [DOCS] Example usage in docstring
As [discussed in #586](https://github.com/koaning/scikit-lego/pull/586#discussion_r1385475426), although most of the library features are documented in the user guide, the best way to showcase how to access and use each class/function would be a minimal example usage in the docstrings. And this is currently covered only for a subset of the library features.

Here the list of all the remaining classes that would benefit from it:

- [ ] `linear_model.LowessRegression`
- [ ] `linear_model.ProbWeightRegression`
- [ ] `linear_model.DeadZoneRegressor`
- [ ] `linear_model.DemographicParityClassifier`
- [ ] `linear_model.EqualOpportunityClassifier`
- [ ] `model_selection.TimeGapSplit`
- [ ] `model_selection.GroupTimeSeriesSplit`
- [ ] `model_selection.KlusterFoldValidation`
- [ ] `naive_bayes.GaussianMixtureNB`
- [ ] `naive_bayes.BayesianGaussianMixtureNB`
- [ ] `neighbors.BayesianKernelDensityClassifier`
- [ ] `decomposition.pca_reconstruction.PCAOutlierDetection`
- [ ] `decomposition.umap_reconstruction.UMAPOutlierDetection`
- [ ] `meta.confusion_balancer.ConfusionBalancer`
- [ ] `meta.estimator_transformer.EstimatorTransformer`
- [ ] `meta.grouped_predictor.GroupedPredictor`
- [ ] `meta.grouped_transformer.GroupedTransformer`
- [x] `meta.outlier_classifier.OutlierClassifier`
- [ ] `meta.regression_outlier_detector.RegressionOutlierDetector`
- [ ] `meta.subjective_classifier.SubjectiveClassifier`
- [ ] `meta.thresholder.Thresholder`
- [ ] `mixture.bayesian_gmm_classifier.BayesianGMMClassifier`
- [ ] `mixture.bayesian_gmm_detector.BayesianGMMOutlierDetector`
- [ ] `mixture.gmm_classifier.GMMClassifier`
- [ ] `mixture.gmm_detector.GMMOutlierDetector`
- [x] `preprocessing.dictmapper.DictMapper`
- [ ] `preprocessing.formulaictransformer.FormulaicTransformer`
- [ ] `preprocessing.identitytransformer.IdentityTransformer`
- [ ] `preprocessing.intervalencoder.IntervalEncoder`
- [x] `preprocessing.outlier_remover.OutlierRemover`
- [ ] `preprocessing.pandastransformers.PandasTypeSelector`
- [ ] `preprocessing.projections.InformationFilter`
- [ ] `preprocessing.repeatingbasis.RepeatingBasisFunction`

As an instance of such minimal example you can refer to [`QuantileRegression` docstring section](https://github.com/koaning/scikit-lego/blob/main/sklego/linear_model.py#L1140), which renders as in its [API section](https://koaning.github.io/scikit-lego/api/linear-model/#sklego.linear_model.QuantileRegression).

If possible try to add **one** unique example covering the relevant features and methods in the top level docstring of the class.

### PR DESCRIPTION
**Pull request by anopsy**
# Description

Added examples to:

- sklego/preprocessing/formulaictransformer.py
- sklego/preprocessing/identitytransformer.py
- sklego/preprocessing/pandastransformers.py
- sklego/preprocessing/projections.py
- sklego/preprocessing/repeatingbasis.py

### CODE REVIEW

**Review by FBruzzesi — COMMENTED** [review]
The difference between the two keywords `Example` and `Examples` is subtle but:

- `Examples` is a keyword in [mkdocstring griffe](https://mkdocstrings.github.io/griffe/docstrings/#numpydoc-section-examples) and has the example as a section, always displayed
- `Example` is a keyword in [mkdocs material](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#+type:example) and adopts the boxed colored layout.

I would opt for the first one as it is numpydoc compliant. However I can see why the second option is also nice and appealing. 
If we go for that, maybe I would suggest to use the 
```
??? example
...
```
block, so that it is closed by default and lowers the clutter in the page, yet easy to access for a user.

@koaning thoughts?

Edit: [collapsible-blocks](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#collapsible-blocks) docs

File: sklego/model_selection.py [hunk 1]
```diff
@@ -260,7 +260,7 @@ def KlusterFoldValidation(**kwargs):
 class ClusterFoldValidation:
     """Cross validator that creates folds based on provided cluster method.
     This ensures that data points in the same cluster are not split across different folds.
-    
+
```
[the remark this answers is not in the document]
Comment (anopsy) [reply sklego/model_selection.py]: Ach so! Thank you for explaining, I also fell for this little test tube in the corner. Just to make sure, the "example" also allows for collapsing, isn't?
Waiting for your decision and will adjust it accordingly

[the remark this answers is not in the document]
Comment (FBruzzesi) [reply sklego/model_selection.py]: Yes, for "example" it is just a matter of default (expanded vs collapsed by default).
Let's wait for Vincent feedback/preference to weight on this 🙃

[the remark this answers is not in the document]
Comment (koaning) [reply sklego/model_selection.py]: Maybe go for usage instead?

[the remark this answers is not in the document]
Comment (FBruzzesi) [reply sklego/model_selection.py]: I thought `Usage` keyword was google-style specific, but at least from the [griffe docs](https://mkdocstrings.github.io/griffe/docstrings/) it seems it is neither google or numpy.

At the end of the day I believe the only difference is that keywords get **bold**-ed (example in screenshot).

I think for now the safest way to proceed is to use `Examples` keyword as it would have the lowest impact and require less changes.

image

[the remark this answers is not in the document]
Comment (anopsy) [reply sklego/model_selection.py]: Okay, I'll do Examples then, if you change your mind, just ping me

**Comment by anopsy** [comment]
I didn't add preprocessing.intervalencoder.IntervalEncoder because I need to delve more into what is it exactly doing. According to the docs strings it's "bending" / "smoothing" the values in X . I checked what "smoothing" in math is doing, and I have my doubts if you meant IntervalEncoder to do  smoothing or you meant a monotonic transformation. If you could give me a hint what to research, I'd be really grateful

**Comment by FBruzzesi** [comment]
@anopsy I hope that the [Interval Encoders user guide]((https://koaning.github.io/scikit-lego/user-guide/preprocessing/#interval-encoders) and the examples there help.

If it doesn't, feel free to reach out to me on socials as well, since user guide would also need some improvements.

Disclaimer: I personally never used the feature
